server/src/main/java/org/elasticsearch/index/fielddata/fieldcomparator/BytesRefFieldComparatorSource.java

```diff
@@ -135,6 +139,11 @@ public void setScorer(Scorable scorer) {
         };
     }
 
+    @Override
+    public BucketedSort newBucketedSort(BigArrays bigArrays, SortOrder sortOrder, DocValueFormat format) {
+        throw new IllegalArgumentException("only supported on numeric values");
```

Comment:
Can we add more information to this error message, e.g. index name, field name, field type?

Reply:
Please ignore if this would make the API horrible due to the need to propagate the required information.